torch/_subclasses/fake_tensor.py

```diff
@@ -726,6 +727,80 @@ def merge_devices(t):
     __torch_function__ = torch._C._disabled_torch_function_impl
 
 
+def _short_circuit_alloc_op(fake_mode, a, b, **kwargs):
+    copy_kwargs = kwargs
+    real_device = copy_kwargs["device"]
+    del copy_kwargs["device"]
+    return FakeTensor(
+        fake_mode,
+        torch.empty_strided(a, b, device="meta", **copy_kwargs),
+        device=real_device,
+    )
+
+
+def _short_circuit_binary_broadcasting_op(a, b):
+    if isinstance(a, (int, float, SymInt, SymFloat)):
+        return b
```

Comment:
not correct, if a is `float`/`SymFloat` and b is an integer tensor, result will be `float`